Cutting with quantum serverless notebook fails with `aiohttp == 3.9.0`

aiohttp 3.9.0 was released a few days ago, and since then CI has been failing because the "notebook" tests fail.  I haven't gotten to the bottom of _why_ they fail, but I have confirmed that a pin fixes things:

```diff
diff --git a/pyproject.toml b/pyproject.toml
index d5bedab..9d1d3a8 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -92,6 +92,7 @@ docs = [
 notebook-dependencies = [
     "circuit-knitting-toolbox[cplex,pyscf]",
     "quantum-serverless>=0.0.7",
+    "aiohttp!=3.9.0",
     "matplotlib",
     "ipywidgets",
     "pylatexenc",
```

Here's the "full" error:

```
=================================== FAILURES ===================================
_ /home/runner/work/circuit-knitting-toolbox/circuit-knitting-toolbox/docs/circuit_cutting/cutqc/tutorials/tutorial_3_cutting_with_quantum_serverless.ipynb _
---------------------------------------------------------------------------
with serverless.context():
    cuts_future = cut_circuit_wires_remote(
        circuit=circuit,
        method="automatic",
        max_subcircuit_width=5,
        max_cuts=2,
        num_subcircuits=[2],
    )
    cuts = get(cuts_future)
---------------------------------------------------------------------------
LocalRayletDiedError                      Traceback (most recent call last)
Cell In[4], line 9
      1 with serverless.context():
      2     cuts_future = cut_circuit_wires_remote(
      3         circuit=circuit,
      4         method="automatic",
   (...)
      7         num_subcircuits=[2],
      8     )
----> 9     cuts = get(cuts_future)

File ~/work/circuit-knitting-toolbox/circuit-knitting-toolbox/.tox/py311-notebook/lib/python3.11/site-packages/quantum_serverless/core/decorators.py:104, in get(object_refs, timeout)
     88 def get(
     89     object_refs: Union[ray.ObjectRef, Sequence[ray.ObjectRef]],
     90     *,
     91     timeout: Optional[float] = None,
     92 ) -> Any:
     93     """Get results from distributed tasks.
     94 
     95     Args:
   (...)
    102         A object or a list of objects.
    103     """
--> 104     return ray.get(object_refs=object_refs,timeout=timeout)

File ~/work/circuit-knitting-toolbox/circuit-knitting-toolbox/.tox/py311-notebook/lib/python3.11/site-packages/ray/_private/auto_init_hook.py:24, in wrap_auto_init.<locals>.auto_init_wrapper(*args, **kwargs)
     21 @wraps(fn)
     22 def auto_init_wrapper(*args, **kwargs):
     23     auto_init_ray()
---> 24     return fn(*args,**kwargs)

File ~/work/circuit-knitting-toolbox/circuit-knitting-toolbox/.tox/py311-notebook/lib/python3.11/site-packages/ray/_private/client_mode_hook.py:103, in client_mode_hook.<locals>.wrapper(*args, **kwargs)
    101     if func.__name__ != "init" or is_client_mode_enabled_by_default:
    102         return getattr(ray, func.__name__)(*args, **kwargs)
--> 103 return func(*args,**kwargs)

File ~/work/circuit-knitting-toolbox/circuit-knitting-toolbox/.tox/py311-notebook/lib/python3.11/site-packages/ray/_private/worker.py:[256](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/actions/runs/6917536253/job/18818830181#step:6:257)5, in get(object_refs, timeout)
   2563             raise value.as_instanceof_cause()
   2564         else:
-> 2565             raise value
   2567 if is_individual_id:
   2568     values = values[0]

LocalRayletDiedError: The task's local raylet died. Check raylet.out for more information.
----------------------------- Captured stderr call -----------------------------
```

If we can't get to the bottom of this very soon, we should merge the pin for the time being.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -92,7 +92,6 @@ docs = [
 notebook-dependencies = [
     "circuit-knitting-toolbox[cplex,pyscf]",
     "quantum-serverless>=0.0.7",
-    "async-timeout",
     "matplotlib",
     "ipywidgets",
     "pylatexenc",

